File: Examples/PHP/Document/RenderPage.php
//ExStart:
<?php
require_once realpath(__DIR__ . '/..') . '/vendor/autoload.php';
require_once realpath(__DIR__ . '/..') . '/Utils.php';

use Aspose\Words\WordsApi;
use Aspose\Words\AsposeApp;

class Document {

	public $wordsApi;

	public function __construct() {
		AsposeApp::$appSID = Utils::appSID;
		AsposeApp::$apiKey = Utils::apiKey;
		$this->wordsApi = new WordsApi();
	}

    public function renderPage()
    {
        // Upload file to Aspose Cloud Storage
        $name = "SampleWordDocument.docx";
        Utils::uploadFile($name);

        $page_index = 1;
        $format = "bmp";

        $result = $this->wordsApi->renderPage($name, $page_index, $format, $storage=null, $folder=null);
    }
}

$document = new Document();
$document->renderPage();

?>
//ExEnd:

File: Examples/Ruby/Document/render_page.rb
#ExStart:
require 'aspose_words_cloud'

class WorkingWithDocument

  include AsposeWordsCloud
  include AsposeStorageCloud

  def initialize
    #Get App key and App SID from https://cloud.aspose.com
    AsposeApp.app_key_and_sid("APP_KEY", "APP_SID")
    @words_api = WordsApi.new  
  end

  def upload_file(file_name)
    @storage_api = StorageApi.new
    response = @storage_api.put_create(file_name, File.open("../../../data/" << file_name,"r") { |io| io.read } )
  end

  def render_page
    file_name = "SampleWordDocument.docx"
    upload_file(file_name)
      
    page_index = 1
    format = "bmp"
    response = @words_api.render_page(file_name, page_index, format, opts = {})
  end

end

workingWithDocument = WorkingWithDocument.new()
puts workingWithDocument.render_page
#ExEnd:
